tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js

```diff
@@ -77,6 +104,91 @@ module.exports = {
       }
       return this.api.page.FilesPageElement.filesList()
     },
+    /**
+     *
+     * @param {String} collaborator
+     * @param {String} requiredPermissions
+     */
+    changeCustomPermissionsTo: async function (collaborator, requiredPermissions) {
+      const requiredPermissionArray = this.permissionArray(requiredPermissions)
+      const informationSelector = util.format(this.elements.collaboratorInformationByCollaboratorName.selector, collaborator)
+      const selectRoleButton = informationSelector + this.elements.selectRoleButtonInCollaboratorInformation.selector
+
+      let permission = ''
+      this
+        .useXpath()
+        .waitForElementVisible(informationSelector)
+        .click(informationSelector)
+        .waitForElementVisible(selectRoleButton)
+
+      for (let i = 0; i < collaboratorPermissionArray.length; i++) {
+        permission = collaboratorPermissionArray[i]
+        const permissionXpath = this.permissionXpathConverter(permission)
+        await this.api.waitForElementVisible({ selector: permissionXpath, timeout: 100, abortOnFailure: false }, function (result) {
+          if (result.value === true) {
```

Comment:
if the permission is given in ` requiredPermissions` but the element is not visible an error should be thrown